transform-plugins/src/main/java/io/cdap/plugin/XMLParser.java

```diff
@@ -77,66 +82,79 @@ public XMLParser(Config config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
-    outSchema = config.getOutputSchema();
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    outSchema = config.getOutputSchema(collector);
+    if (!this.config.containsMacro(INPUT)
+        && pipelineConfigurer.getStageConfigurer().getInputSchema().getField(this.config.inputField) == null) {
+      collector.addFailure(String.format("Input field '%s' must exist in input schema.", this.config.inputField), null)
+          .withConfigProperty(INPUT);
+    }
     try {
-      validateXpathAndSchema();
+      validateXpathAndSchema(collector);
     } catch (UnsupportedEncodingException e) {
```

Comment:
whatever can throw this exception should be handled within the validate method and not out here

Reply:
Fixed